plugins/in_http/http_prot.c

```diff
@@ -510,21 +526,71 @@ static int process_payload(struct flb_http *ctx, struct http_conn *conn,
         type = HTTP_CONTENT_URLENCODED;
     }
 
+    /* handle gzip-encoded content */
+    /* 
+        The way things come out of headers_extra is very strange; everything seems to be in
+        key->data of the first entry and val->data is everything minus the first header key.
+        The behavior of monkey seems to be pretty buggy here.
+
+        ...so, we just grab the whole thing and then sift through it looking for content-encoding.
+    */
+    if (session->parser.headers_extra_count > 0) {
+        extra_headers = &session->parser.headers_extra[0];
+        const char *content_encoding = "content-encoding";
+        if (extra_headers->key.data != NULL) {
+            extracted_extra_headers = mk_string_tolower(extra_headers->key.data);
+            content_encoding_header_location = mk_string_search(extracted_extra_headers, content_encoding, 0);
+        }
+    }
+    
+    if (content_encoding_header_location == -1) {
+        uncompressed_data = request->data.data;
+        uncompressed_size = request->data.len;
+    } else {
+        /* Extract content-encoding and everything after up to the next \r\n to find the header value */
+        extracted = mk_string_copy_substr(extracted_extra_headers, content_encoding_header_location, strlen(extracted_extra_headers));
+        mk_mem_free(extracted_extra_headers);
+        first_colon = mk_string_search(extracted, ":", 0);
+        first_newline = mk_string_search(extracted, "\r\n", 0);
+        
+        content_encoding_header_value = mk_string_copy_substr(extracted, first_colon+1, first_newline);
+        mk_mem_free(extracted);
+        mk_string_trim(&content_encoding_header_value);
+        content_encoding_compare = strncasecmp(content_encoding_header_value, "gzip", 5);
```

Comment:
How about **gzip_compressed** like flag variable instead of using common named variable here?